<a href="https://colab.research.google.com/github/porkostomus/jupyter/blob/master/completing_the_square.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Rewrite the equation by completing the square.**

$x^2 + x -72=0$

$x^2 + x=72$

In [57]:
from sympy import *
x = Symbol('x')

def rewriteeq(coef1, coef2, const):
    return factor(Eq(coef1*x**2 + coef2*x + Rational(coef2,2)**2, 0 - const + Rational(coef2,2)**2))

rewriteeq(1, 1, -72)

Eq((2*x + 1)**2/4, 289/4)

$x^2+x + {\dfrac{1}{4}}=\left(x +\dfrac{1}{2} \right)^2$

$x^2 + x=72$

$x^2 + x + {\dfrac{1}{4}}=72 + {\dfrac{1}{4}}$

$\left(x +\dfrac{1}{2}\right)^2=\dfrac{289}{4}$

**Rewrite the function by completing the square.**

$f(x)= x^{2} + x -30$

In [56]:
def rewritefn(coef1, coef2, const):
    return factor(coef1*x**2 + coef2*x + Rational(coef2,2)**2) + const - Rational(coef2,2)**2

rewritefn(1, 1, -30)

(2*x + 1)**2/4 - 121/4

$\left(x^2 +x\right) -30$

$x^2+x+{\dfrac{1}{4}}=\left(x +\dfrac{1}{2}\right)^2$

Add ${\dfrac{1}{4}}$ inside the parentheses, and subtract ${1}\cdot{\dfrac{1}{4}}$ outside them, to keep the expression equivalent.

$\left(x^2 + x\right) -30$

$\left(x^2 +x+{\dfrac{1}{4}}\right) -30 -{1}\cdot{\dfrac{1}{4}}$

$\left(x +\dfrac{1}{2}\right)^2 -30 -\dfrac{1}{4}$

$\left(x +\dfrac{1}{2}\right)^2 -\dfrac{121}{4}$

**Rewrite the equation by completing the square.**

$2x^{2}-3x-5 = 0$

$2x^2-3x=5$

$x^2-\dfrac32x=\dfrac52$

Now we want to complete $x^2-\dfrac32x$ into a perfect square.

To do that, we should add $\left(\dfrac{-\frac32}{2}\right)^2={\dfrac{9}{16}}$ to it:

$x^2{-\dfrac32}x+{\dfrac{9}{16}}=\left(x-\dfrac34\right)^2$

$x^2-\dfrac32x=\dfrac52$

$x^2-\dfrac32x+{\dfrac{9}{16}}$

$\dfrac52+{\dfrac{9}{16}}$

$\left(x-\dfrac34\right)^2=\dfrac{49}{16}$


$f(x)= 4 x^{2} +12 x +9$

$4 \left(x^2 +3 x\right) +9$

Now we want to complete $x^2 +3 x$ into a perfect square.

To do that, we should add $\left(\dfrac{3}{2}\right)^2={\dfrac{9}{4}}$ to it:

$x^2{+3}x+{\dfrac{9}{4}}=\left(x +\dfrac{3}{2}\right)^2$

We add ${\dfrac{9}{4}}$ inside the parentheses, and subtract ${4}\cdot{\dfrac{9}{4}}$ outside them, to keep the expression equivalent.

$4 \left(x^2 +3 x\right) +9$

$4\left(x^2 +3 x+{\dfrac{9}{4}}\right) +9 -{4}\cdot{\dfrac{9}{4}}$

\\\\ &=4 \left(x +\dfrac{3}{2}\right)^2 +9 -9 \\\\ &=4 \left(x +\dfrac{3}{2}\right)^2 +0 \end{aligned} 
​	  
=4(x 
2
 +3x)+9
=4(x 
2
 +3x+ 
4
9
​	 )+9−4⋅ 
4
9
​	 
=4(x+ 
2
3
​	 ) 
2
 +9−9
=4(x+ 
2
3
​	 ) 
2
 +0
​	 
Hint #44 / 4
In conclusion, the function after completing the square is written as:
f(x)=4 \left(x +\dfrac{3}{2}\right)^2 +0f(x)=4(x+ 
2
3
​	 ) 
2
 +0

**Rewrite the equation by completing the square.**

$x^{2}+4x-21 = 0$

Begin by moving the constant term to the right side of the equation.

$x^2 + 4x = 21$

In [8]:
from sympy import *
x = Symbol('x')

coef = 4
const = -21
eq = Eq(x**2 + coef*x + const, 0)
eq2 = Eq(x**2 + coef*x, 0 - const)
eq2

Eq(x**2 + 4*x, 21)

We complete the square by taking half of the coefficient of our $x$ term, squaring it, and adding it to both sides of the equation. Since the coefficient of our $x$ term is $4$, half of it would be $2$, and squaring it gives us $4$.

$x^2 + 4x + 4 = 21 + 4$

In [14]:
eq3 = Eq(x**2 + coef*x + Rational(coef,2)**2, 0 - const + Rational(coef,2)**2) 
eq3

Eq(x**2 + 4*x + 4, 25)

**We can now rewrite the left side of the equation as a squared term.**

$( x + 2 )^2 = 25$

In [15]:
factor(eq3)

Eq((x + 2)**2, 25)

Rewrite the equation by completing the square.

$x^{2}+2x-48 = 0$

Begin by moving the constant term to the right side of the equation.

$x^2 + 2x = 48$

In [16]:
from sympy import *
x = Symbol('x')

coef = 2
const = -48
eq = Eq(x**2 + coef*x + const, 0)
eq2 = Eq(x**2 + coef*x, 0 - const)
eq2

Eq(x**2 + 2*x, 48)

We complete the square by taking half of the coefficient of our $x$ term, squaring it, and adding it to both sides of the equation. Since the coefficient of our $x$ term is $2$, half of it would be $1$, and squaring it gives us $1$.

$x^2 + 2x + 1 = 48 + 1$

We can now rewrite the left side of the equation as a squared term.

$( x + 1 )^2 = 49$

In [17]:
eq3 = Eq(x**2 + coef*x + Rational(coef,2)**2, 0 - const + Rational(coef,2)**2) 
factor(eq3)

Eq((x + 1)**2, 49)

**Rewrite the equation by completing the square.**

$x^{2}-2x-35 = 0$

In [34]:
def rewriteeq(coef, const):
    return factor(Eq(x**2 + coef*x + Rational(coef,2)**2, 0 - const + Rational(coef,2)**2))

rewriteeq(-2, -35)

Eq((x - 1)**2, 36)

**Rewrite the function by completing the square.**

$f(x)=x^{2}-12x+50$

In [46]:
rewritefn(1, -12, 50)

(x - 6)**2 + 49


$f(x)=x^{2}+2x-39$

In [33]:
coef = -12
const = 50
factor(x**2 + coef*x + Rational(coef,2)**2) + const - Rational(coef,2)**2

(x + 1)**2 - 40

$x^{2}+2x-48 = 0$

In [45]:
rewriteeq(1, 2, -48)

Eq((x + 1)**2, 49)

$x^{2}+2x-3 = 0$

In [47]:
rewriteeq(1, 2, -3)

Eq((x + 1)**2, 4)

$x^{2}-8x+7 = 0$

In [59]:
rewriteeq(1, -8, 7)

Eq((x - 4)**2, 9)

$f(x)=x^{2}-8x-2$

In [58]:
rewritefn(1, -8, -2)

(x - 4)**2 - 18

$f(x)=x^{2}+2x+26$

In [60]:
rewritefn(1, 2, 26)

(x + 1)**2 + 25

$x^{2}+14x+49 = 0$

In [61]:
rewriteeq(1, 14, 49)

Eq((x + 7)**2, 0)

$x^{2}-14x+40 = 0$

In [62]:
rewriteeq(1, -14, 40)

Eq((x - 7)**2, 9)

$f(x)=x^{2}-6x-60$

In [63]:
rewritefn(1, -6, -60)

(x - 3)**2 - 69

$f(x)=x^{2}+20x+40$

In [64]:
rewritefn(1, 20, 40)

(x + 10)**2 - 60